pymagicc/core.py

```diff
@@ -258,7 +263,11 @@ def run(self, scenario=None, only=None, **kwargs):
         try:
             run_paras = self.read_parameters()
             self.config = run_paras
-            mdata.metadata["parameters"] = run_paras
+            try:
```

Comment:
This is the pythonic way of doing things - asking for forgiveness (exceptions) instead of permission (explicit checks), but it isn't that clear. Perhaps a change is needed to be more explicit instead of an `UnboundLocalError`